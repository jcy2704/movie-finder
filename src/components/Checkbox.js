/* eslint-disable react/jsx-boolean-value */
import React from 'react';
import PropTypes from 'prop-types';

const Checkbox = ({
  id, name, handleChange, checked,
}) => (
  <div className="each-filter col">
    <input type="checkbox" value={name} id={id} onChange={handleChange} defaultChecked={checked(id.toString())} />
    <label htmlFor={name}>{name}</label>
  </div>
);

Checkbox.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  checked: PropTypes.func.isRequired,
};

export default Checkbox;
