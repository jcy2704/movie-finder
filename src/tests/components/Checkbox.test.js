import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from '../../components/Checkbox';

it('Renders Checkbox', () => {
  const popular = renderer
    .create(
      <Checkbox checked={() => true} handleChange={() => ''} id={1} name="action" />,
    )
    .toJSON();

  expect(popular).toMatchSnapshot();
});
