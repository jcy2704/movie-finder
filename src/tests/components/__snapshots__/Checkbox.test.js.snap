// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Checkbox 1`] = `
<div
  className="each-filter col"
>
  <input
    defaultChecked={true}
    id={1}
    onChange={[Function]}
    type="checkbox"
    value="action"
  />
  <label
    htmlFor="action"
  >
    action
  </label>
</div>
`;
